public/app/plugins/panel/bargauge/BarGaugePanel.tsx

```diff
@@ -45,17 +45,17 @@ export class BarGaugePanel extends PureComponent<PanelProps<BarGaugeOptions>> {
     const { value } = valueProps;
     const { hasLinks, getLinks } = value;
 
-    if (!hasLinks) {
-      return this.renderComponent(valueProps, {});
+    if (hasLinks && getLinks) {
+      return (
+        <DataLinksContextMenu links={getLinks}>
+          {api => {
+            return this.renderComponent(valueProps, api);
+          }}
+        </DataLinksContextMenu>
+      );
     }
 
-    return (
-      <DataLinksContextMenu links={getLinks}>
-        {api => {
-          return this.renderComponent(valueProps, api);
-        }}
-      </DataLinksContextMenu>
-    );
+    return this.renderComponent(valueProps, {});
```

Comment:
Interesting change, out of curiosity why? Is this more specific than previous?

Reply:
Was on a fence about this change TBH. `getLinks` can be undefined. It's also lazily evaluated so we would have to run the interpolation twice to know if there are links or not. I think that's why the `hasLinks` was added. So, in order for the `DataLinksContextMenu` no to yield about passing possible undefined, I have reversed the condition. 

![giphy](https://user-images.githubusercontent.com/2376619/84655461-4418c400-af11-11ea-8cd1-8d26a10e8875.gif)